src/AccessibilityInsights/MainWindowHelpers/Initialize.cs

```diff
@@ -411,45 +411,83 @@ private Action InvokeAction(Action action)
         }
 
         /// <summary>
-        /// Sets window position to avoid opening window offscreen. Modelled after explorer's behavior
-        /// If fully offscreen, open in default position
-        /// If partially offscreen, move onscreen
+        /// Make sure that the app layout is fully within the virtual screen, including resizing it
+        /// if necessary.
         /// </summary>
-        /// <returns></returns>
-        private void FixWindowPositionForStartup()
+        /// <param name="layout">The layout. Its position will be modified only if necessary</param>
+        private void EnsureWindowIsInVirtualScreen(AppLayout layout)
+        {
+            EnsureWindowIsInVirtualScreenWithInjection(layout, this.Top, this.Left,
+                SystemParameters.VirtualScreenLeft, SystemParameters.VirtualScreenTop,
+                SystemParameters.VirtualScreenWidth, SystemParameters.VirtualScreenHeight);
+        }
+
+        /// <summary>
+        /// Return the first non-NaN number from an array of potentialNumbers.
+        /// Will return 0.0 if they're all NaN.
+        /// </summary>
+        private static double GetFirstNumber(double[] potentialNumbers)
+        {
+            foreach(double potentialNumber in potentialNumbers)
+            {
+                if (!double.IsNaN(potentialNumber))
+                {
+                    return potentialNumber;
+                }
+            }
+
+            return 0.0;
+        }
+
+        internal static void EnsureWindowIsInVirtualScreenWithInjection(AppLayout layout,
+            double windowTop, double windowLeft, double virtualLeft,
+            double virtualTop, double virtualWidth, double virtualHeight)
         {
-            var layout = ConfigurationManager.GetDefaultInstance().AppLayout;
+            double virtualRight = virtualLeft + virtualWidth;
+            double virtualBottom = virtualTop + virtualHeight;
 
             if (layout != null)
             {
+                layout.Top = GetFirstNumber(new double[] { layout.Top, windowTop, 200.0 });
+                layout.Left = GetFirstNumber(new double[] { layout.Left, windowLeft, 200.0 });
```

Comment:
Am I correct in understanding that these 200.0s are fallback values in case of NaNs? If so, this might benefit from a comment or pulling out as a const like `fallbackWindowPosition =  200.0`